lib/auth.js

```diff
@@ -48,6 +55,66 @@ Auth.prototype.bearer = function (bearer, sendImmediately) {
   }
 }
 
+
+Auth.prototype.negotiate = function(request, authHeader, sendImmediately) {
+  var self = this
+
+  if (kerberos === undefined) {
+    // Nothing to do if we don't have Kerberos
+    // or a current credential cache
+    return
+  }
+
+  self.hasAuth = true
+
+  if (!sendImmediately) {
+    return
+  }
+
+  var k = new kerberos.Kerberos()
+  var service_identifier = 'HTTP@' + request.host
+
+  self.pendingAuth = true
+  k.authGSSClientInit(service_identifier,
+    kerberos.Kerberos.GSS_C_MUTUAL_FLAG,
+    function (err, context) {
+      if (err) {
+        self.request.emit('error', new Error('auth() error with Kerberos: ' + err))
+        return
+      }
+
+      // Pass in the challenge data if any from the
+      // remote host
+      k.authGSSClientStep(context, authHeader || '', function(err, result) {
+        if (err) {
+          self.request.emit('error', new Error('auth() error with Kerberos: ' + err))
+          return
+        }
+
+        var content = context.response
+
+        k.authGSSClientClean(context, function(err) {
+          if (err) {
+            self.request.emit('error', new Error('auth() error with Kerberos: ' + err))
+            return
+          }
+
+          request.setHeader('Authorization', 'Negotiate ' + content)
```

Comment:
That might be true, I just wanted to point it out, it's not the most urgent thing.